File: neetcode/src/test/java/solutions/arraysandhashing/ProductOfArrayExceptSelfTest.java
package solutions.arraysandhashing;

import static org.junit.Assert.assertTrue;
import org.junit.Test;
import static solutions.arraysandhashing.ProductOfArrayExceptSelf.productExceptSelf;

import java.util.Arrays;

public class ProductOfArrayExceptSelfTest{
  @Test
  public void example1(){
    // initialize
    int[] input = {1,2,3,4};
    int[] expected = {24,12,8,6};
    int[] actual = productExceptSelf(input);

    // test
    String msg = "Expected " + Arrays.toString(expected) + 
      ", returned " + Arrays.toString(actual);
    assertTrue(msg, Arrays.equals(expected, actual));
  }

  @Test
  public void example2(){
    // initialize
    int[] input = {-1,1,0,-3,3};
    int[] expected = {0,0,9,0,0};
    int[] actual = productExceptSelf(input);

    // test
    String msg = "Expected " + Arrays.toString(expected) + 
      ", returned " + Arrays.toString(actual);
    assertTrue(msg, Arrays.equals(expected, actual));
  }
}

File: neetcode/python/test/arraysandhashing/product_of_array_except_self_test.py
import unittest
from typing import List
from src.arraysandhashing.product_of_array_except_self import Solution

class ProductOfArrayExceptSelfTest(unittest.TestCase):
    def test_example1(self):
        nums = [1,2,3,4]
        actual = Solution.productExceptSelf(nums)
        expected = [24,12,8,6]
        self.assertEqual(actual, expected)

    def test_example2(self):
        nums = [-1,1,0,-3,3]
        actual = Solution.productExceptSelf(nums)
        expected = [0,0,9,0,0]
        self.assertEqual(actual, expected)
